plugins/filter_lua/lua.c

```diff
@@ -112,6 +112,45 @@ static void mpack_buffer_flush(mpack_writer_t* writer, const char* buffer, size_
     flb_sds_cat_safe(&ctx->packbuf, buffer, count);
 }
 
+static void pack_result_mpack(lua_State *l,
+                              mpack_writer_t *writer,
+                              struct flb_lua_l2c_config *l2cc,
+                              struct flb_time *t)
+{
+  int i;
+  int len;
+
+  if (lua_type(l, -1) != LUA_TTABLE) {
+    return;
+  }
+
+  len = flb_lua_arraylength(l);
+  if (len > 0) {
+    /* record split */
+    for (i = 1; i <= len; i++) {
+      /* write array tag */
+      mpack_write_tag(writer, mpack_tag_array(2));
+      /* write timestamp */
+      flb_time_append_to_mpack(writer, t, 0);
+      /* get the subrecord */
+      lua_rawgeti(l, -1, i);
+      /* convert */
+      flb_lua_tompack(l, writer, 0, l2cc);
+      lua_pop(l, 1);
+    }
+  }
+  else {
+      /* write array tag */
+      mpack_write_tag(writer, mpack_tag_array(2));
+      /* write timestamp */
+      flb_time_append_to_mpack(writer, t, 0);
+      /* convert */
+      flb_lua_tompack(l, writer, 0, l2cc);
+  }
+  /* pop */
+  lua_pop(l, 1);
+}
```

Comment:
Could you check indent level ?

https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#coding-style
```
You have to pay attention to the code indentation, tabs are 4 spaces, spaces on conditionals, etc. If your code submission is not aligned, it will be rejected.
```